### ISSUE TITLE
new nixos-rebuild switch hang forever

### ISSUE THREAD

**Issue by sg-qwt** [NixOS/nixpkgs#353450]
After upgrading to latest unstable (7ffd9ae656aec493492b44d0ddfb28e79a1ea25d), I found my system is using the the ng nixos-rebuild switch. Which always hang indefinitely without any meaningful error messages.

```
building the system configuration...
activating the configuration...
sops-install-secrets: Imported /etc/ssh/ssh_host_ed25519_key as age key with fingerprint xxx
setting up /etc...
sops-install-secrets: Imported /etc/ssh/ssh_host_ed25519_key as age key with fingerprint xxx
Setting up sops templates...
reloading user units for me...
restarting sysinit-reactivation.target
```
It always stuck on restarting sysinit-reactivation.target.

journalctl -u nixos-rebuild-switch-to-configuration.service
```
Nov 04 01:30:51 ge systemd[1]: Starting /nix/store/sf6y4arqcm100rnnl3dhpg732i774zp6-coreutils-9.5/bin/true...
Nov 04 01:30:51 ge systemd[1]: Started /nix/store/sf6y4arqcm100rnnl3dhpg732i774zp6-coreutils-9.5/bin/true.
Nov 04 01:30:51 ge systemd[1]: nixos-rebuild-switch-to-configuration.service: Deactivated successfully.
Nov 04 01:30:51 ge systemd[1]: Starting /nix/store/4ybal6rxnh3slnz644iv769b1374w88k-nixos-system-ge-24.11.20241102.7ffd9ae/bin/switch-to-configuration switch...
Nov 04 01:30:51 ge systemd[1]: Started /nix/store/4ybal6rxnh3slnz644iv769b1374w88k-nixos-system-ge-24.11.20241102.7ffd9ae/bin/switch-to-configuration switch.
Nov 04 01:30:52 ge nixos[2600]: switching to system configuration /nix/store/4ybal6rxnh3slnz644iv769b1374w88k-nixos-system-ge-24.11.20241102.7ffd9ae
```

How do I troubleshot this further this case?

In #313696 someone proposed `services.tcsd.enable = false;`, but I guess that's only specific to his case, I don't have TPM or tcsd enabled at all.

Also if I just restart my pc, I can see the new generation is there as expected in the boot menu, so I believe everything just worked as expected but somehow switch just hang for some reason forever?

**Comment by Frontear** [comment]
Can you rollback to the previous working generation, disable the switch-ng, and rebuild on the same nixpkgs revision, then once again test if it hangs in the new generation?

**Comment by Frontear** [comment]
cc @jmbaur as maintainer of stc-ng

**Comment by sg-qwt** [comment]
I've found a weird pattern, only the first switch after fresh reboot can success without hanging forever.
Even if I do an immediate `nixos-rebuild switch` with no code change, it hangs regardless of enableNg.
![image](https://github.com/user-attachments/assets/d3df6e53-4cd6-4331-9063-d253e703a738)

**Comment by sg-qwt** [comment]
After some help from @Aleksanaa , I've found my culprit. Thanks.

Troubleshoot steps:
Run `systemctl`, I've found tailscaled-autoconnect.service with a dot. Tailscale failed to register due to key expires. After I disabled tailscale service, switch can complete without hanging anymore.

However, I remember clearly that tailscaled-autoconnect just a oneshot service expected to fail. I've had previous failure with that tailscale registering as well, but that shall not block the whole system switch to hang. Previously, I remember clearly that on first system switch would correctly report service failure and continue without hanging forever. Then on every following switch, it will not bugged by that one-shot failure and continue switch without any problem.

Now with that one failure service, only the first system switch complain and switch correctly, but I do not expect system switch afterwards to hang forever due to that service, also there's no error or any helpful information to tell me why it hangs this case. 

cc @Frontear @jmbaur
